cluster-agent/controllers/argoproj.io/application_controller_test.go

```diff
@@ -566,6 +582,229 @@ var _ = Describe("Application Controller", func() {
+			}
+			guestbookApp.Status.Conditions = append(guestbookApp.Status.Conditions, app)
+
+			By("Add databaseID label to applicationID")
+			databaseID := guestbookApp.Labels[dbID]
+			applicationDB := &db.Application{
+				Application_id:          databaseID,
+				Name:                    name,
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			applicationStateDB := &db.ApplicationState{
+				Applicationstate_application_id: applicationDB.Application_id,
+			}
+
+			By("Create Application in Database")
+			err = reconciler.DB.CreateApplication(ctx, applicationDB)
+			Expect(err).To(BeNil())
+
+			By("Verify SyncError is not present in database")
+			err = reconciler.DB.GetApplicationStateById(ctx, applicationStateDB)
+			Expect(err).ToNot(BeNil())
+			Expect(db.IsResultNotFoundError(err)).To(BeTrue())
+
+			By("Create a new ArgoCD Application")
+			err = reconciler.Create(ctx, guestbookApp)
+			Expect(err).To(BeNil())
+
+			By("Call reconcile function")
+			result, err := reconciler.Reconcile(ctx, newRequest(namespace, name))
+			Expect(err).To(BeNil())
+			Expect(result).ToNot(BeNil())
+
+			err = reconciler.DB.GetApplicationStateById(ctx, applicationStateDB)
+			Expect(applicationStateDB.ReconciledState).ToNot(BeNil())
+			Expect(err).To(BeNil())
+
+			// Checking for syncError value in database object
+			for _, syncError := range guestbookApp.Status.Conditions {
+				Expect(applicationStateDB.SyncError).To(BeEmpty())
+				Expect(syncError.Type).ToNot(Equal("SyncError"))
+			}
+		})
+
+		It("Test to check whether existing value in syncError field is updated to new value when conditions.Message of ArgoCD Application changes", func() {
+			By("Close database connection")
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+
+			ctx = context.Background()
+
+			By("Add databaseID label to applicationID")
+			databaseID := guestbookApp.Labels[dbID]
+			applicationDB := &db.Application{
+				Application_id:          databaseID,
+				Name:                    name,
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			reconciledStateString, reconciledObj, err := dummyApplicationComparedToField()
+			Expect(reconciledObj.Destination.Namespace).ToNot(Equal(guestbookApp.Status.Sync.ComparedTo.Destination.Namespace))
+			Expect(err).To(BeNil())
+
+			// applicationState which already exists in database
+			applicationState := &db.ApplicationState{
+				Applicationstate_application_id: applicationDB.Application_id,
+				Health:                          "Healthy",
+				Sync_Status:                     "Synced",
+				ReconciledState:                 reconciledStateString,
+				SyncError:                       "test-sync-error",
```

Comment:
We are directly updating the sync error message in the DB object. Shouldn't we update it in the Argo CD Application?

Reply:
Here the test is as follows, where  SyncError is already present in db  which is "test-sync-error" and once reconcile function is called we need to check whether db is updated with the syncError similar to the ArgoCD Application syncError